docs/Internals.md

```diff
@@ -1,150 +1,165 @@
-  void store(String key, P payload, Persistence<D> dataPersistence);
-
-  void close();
-}
-```
-  
-For the [store/load](../README.md#store--load) usage model, we also need to implement the `Persistence` interface
-
-```java
-// When using the store/load style, this defines the callbacks used to interact with Data Row Records.
-interface Persistence<T> {
-  Optional<T> load(String key);
-  String store(T value);
-  void store(String key, T value);
-  String generateKey(T value);
-}
-```
-
-### Crypto Policy
-
-```java
-  // Used to configure various behaviors of the internal algorithm
-  interface CryptoPolicy {
-    enum KeyRotationStrategy {
-        INLINE, // This is the only one currently supported/implemented
-        QUEUED
-    };
-    KeyRotationStrategy keyRotationStrategy();
-
-    boolean isKeyExpired(Instant keyCreationDate);
-    long getRevokeCheckPeriodMillis();
-
-    boolean canCacheSystemKeys();
-    boolean canCacheIntermediateKeys();
-
-    boolean notifyExpiredIntermediateKeyOnRead();
-    boolean notifyExpiredSystemKeyOnRead();
-}
-```
-
-Detailed information about the CryptoPolicy can be found [here](CryptoPolicy.md) 
-
-### Metastore
-
-```java
-  // Defines the backing metastore
-  interface MetastorePersistence<V> {
-    Optional<V> load(String keyId, Instant created);
-    Optional<V> loadLatestValue(String keyId);
-
-    boolean store(String keyId, Instant created, V value);
-}
-```
-
-Detailed information about the Metastore can be found [here](Metastore.md) 
-
-
-### Key Management Service
-
-```java
-  // Defines the root KMS
-  interface KeyManagementService {
-    byte[] encryptKey(CryptoKey key);
-    CryptoKey decryptKey(byte[] keyCipherText, Instant keyCreated, boolean revoked);
-
-    <T> T withDecryptedKey(byte[] keyCipherText, Instant keyCreated, boolean revoked,
-                           BiFunction<CryptoKey, Instant, T> actionWithDecryptedKey);
-}
-```
-
-Detailed information about the  Key Management Service can be found [here](KeyManagementService.md) 
-
+# SDK Internals
+
+## Algorithm Internals
+
+### Encrypt 
+Depending on policy, we will either continue to encrypt if a key in the tree has expired or rotate/generate keys inline.
+
+    Data is ready to write to data persistence
+    If latest IK is not cached or latest IK in cache is expired
+        Load latest IK EKR from metadata persistence
+        If IK is found
+            If IK is not expired or (IK is expired and policy allows queued rotation)
+                If SK is not cached
+                    Load specific SK EKR from metadata persistence
+                    If SK EKR DOES NOT exist in metadata persistence
+                        Fall through to new IK creation
+                    If allowed by policy, add SK to protected memory cache
+                If SK is expired
+                    # NOTE: Possible inconsistency: when policy doesn't use inline rotation, consider proceeding without forced creation (same as IK handling)
```

Comment:
minor: 120 char line limit (may be another line or two in this file violating the limit, one of which i think i goofed)